src/graph_notebook/notebooks/04-Machine-Learning/Neptune-ML-SPARQL/neptune_ml_sparql_utils.py

```diff
@@ -0,0 +1,701 @@
+
+        s3_client = boto3.client('s3')
+        for root, dirs, files in os.walk(self.formatted_directory):
+            for file in files:
+                s3_client.upload_file(os.path.join(
+                    self.formatted_directory, file), bucket, f'{file_path}/{file}')
+
+    def prepare_movielens_data_rdf(self, s3_bucket: str, clear_staging_area: bool):
+        self.formatted_directory = f'{self.formatted_directory}/rdf'
+        if clear_staging_area:
+            print('clearing staging area by default, use "clear_staging_area=False" to retain')
+            shutil.rmtree(self.formatted_directory, ignore_errors=True)
+            os.makedirs(self.formatted_directory, exist_ok=True)
+        bucket_name = f'{s3_bucket}/neptune-formatted/movielens-100k/rdf/'
+        self.__download_and_unzip()
+        self.__process_movies_genres_rdf()
+        self.__process_users_rdf()
+        self.__process_ratings_users_rdf()
+        self.__upload_to_s3(bucket_name)
+        if clear_staging_area:
+            shutil.rmtree(self.formatted_directory, ignore_errors=True)
+        print('Completed Processing, data is ready for loading using the s3 url below:')
+        print(bucket_name)
+        return bucket_name
+
+    def __process_movies_genres_rdf(self):
+        # process the movies_vertex.csv
+        print('Processing Movies to RDF')
+        movies_df = pd.read_csv(os.path.join(
+            self.raw_directory, 'ml-100k/u.item'), sep='|', encoding='ISO-8859-1',
+            names=['~id', 'title', 'release_date', 'video_release_date', 'imdb_url',
+                   'unknown', 'Action', 'Adventure', 'Animation', 'Childrens', 'Comedy',
+                   'Crime', 'Documentary', 'Drama', 'Fantasy', 'Film-Noir', 'Horror', 'Musical',
+                   'Mystery', 'Romance', 'Sci-Fi', 'Thriller', 'War', 'Western'])
+        # Parse date and convert to ISO format
+        movies_df['release_date'] = movies_df['release_date'].apply(
+            lambda x: str(
+                datetime.strptime(x, '%d-%b-%Y').isoformat()) if not pd.isna(x) else x)
+        movies_df['~label'] = 'movie'
+        movies_df['~id'] = movies_df['~id'].apply(
+            lambda x: f'movie_{x}')
+
+        genres = ['unknown', 'Action', 'Adventure', 'Animation', 'Childrens', 'Comedy',
+                  'Crime', 'Documentary', 'Drama', 'Fantasy', 'Film-Noir', 'Horror', 'Musical',
+                  'Mystery', 'Romance', 'Sci-Fi', 'Thriller', 'War', 'Western']
+
+        genre_df = pd.DataFrame(genres, columns=['~id'])
+        genre_df['~label'] = 'genre'
+        genre_df['name'] = genre_df['~id']
+
+        movie_rdf_filename = os.path.join(self.formatted_directory, 'movies.nq')
+        movie_genre_rdf_filename = os.path.join(self.formatted_directory, 'movie_genres.nq')
+        movie_graph = ConjunctiveGraph()
+        movie_genre_graph = ConjunctiveGraph()
+
+        # movie vertex file creation
+        for index, row in movies_df.iterrows():
```

Comment:
Since you aren't using the index here you don't need `iterrows()` you can just do `for row in movies_df`

Reply:
Removing this line resulted in an error, so i have kept it for now.